src/main/java/uk/gov/hmcts/reform/sscs/service/LetterUtils.java

```diff
@@ -197,4 +191,136 @@ public static Stream<Pair<String, YesNo>> buildOtherPartiesForReasonableAdjustme
         }
         return otherPartyReasonableAdjustmentList.stream();
     }
+
+    public static String getNameForSender(SscsCaseData sscsCaseData) {
+        if (nonNull(sscsCaseData.getOriginalSender())) {
+            final String originalSenderCode = sscsCaseData.getOriginalSender().getValue().getCode();
+            if (originalSenderCode.equalsIgnoreCase("appellant")) {
+                return sscsCaseData.getAppeal().getAppellant().getName().getFullNameNoTitle();
+            } else if (originalSenderCode.equalsIgnoreCase("representative")) {
+                return SendNotificationHelper.getRepSalutation(sscsCaseData.getAppeal().getRep(), false);
+            } else if (originalSenderCode.equalsIgnoreCase("jointParty")) {
+                return sscsCaseData.getJointParty().getName().getFullNameNoTitle();
+            } else {
+                return getOtherPartyName(sscsCaseData)
+                        .map(Name::getFullNameNoTitle).orElse("");
+            }
+        }
+        return "";
+    }
+
+    private static Optional<Representative> getRepresentativeOfOtherParty(SscsCaseData sscsCaseData) {
+        for (CcdValue<OtherParty> op : sscsCaseData.getOtherParties()) {
+            if (op.getValue().hasRepresentative()) {
+                boolean isValidRepresentative = sscsCaseData.getOriginalSender().getValue().getCode().contains(op.getValue().getRep().getId());
+                if (isValidRepresentative) {
+                    return Optional.of(op.getValue().getRep());
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<Name> getOtherPartyName(SscsCaseData sscsCaseData) {
+        boolean otherPartyRep = sscsCaseData.getOriginalSender().getValue().getCode().contains("otherPartyRep");
+        if (otherPartyRep) {
+            Optional<Representative> representative = getRepresentativeOfOtherParty(sscsCaseData);
+            return representative.isPresent() ? Optional.of(representative.get().getName())
+                    : Optional.empty();
+        } else {
+            return sscsCaseData.getOtherParties().stream()
+                    .filter(op -> sscsCaseData.getOriginalSender().getValue().getCode().contains(op.getValue().getId()))
+                    .findFirst()
+                    .map(op -> op.getValue().getName());
+        }
+    }
```

Comment:
```suggestion
  public static Optional<Name> getOtherPartyName(SscsCaseData sscsCaseData) {
    String originalSenderCode = sscsCaseData.getOriginalSender().getValue().getCode();
    boolean otherPartyRep = originalSenderCode.contains("otherPartyRep");

    return otherPartyRep
            ? getRepresentativeOfOtherParty(sscsCaseData).map(Representative::getName)
            : sscsCaseData.getOtherParties().stream()
                    .filter(op -> originalSenderCode.contains(op.getValue().getId()))
                    .findFirst()
                    .map(op -> op.getValue().getName());
}
```

Reply:
I understand your enthusiasm but its too complex ternary operator for return statement :D @amoslv1

Reply:
Maybe converting the ternary into an if would be ok?